src/Lean/Meta/Tactic/Simp/Types.lean

```diff
@@ -97,6 +99,19 @@ def getSimpCongrTheorems : M SimpCongrTheorems :=
   finally
     modify fun s => { s with cache := cacheSaved }
 
+def recordSimpTheorem (n : Name) : SimpM Unit :=
+  modify fun s => { s with usedTheorems := s.usedTheorems.insert n }
+
+def ppSimpTheorem (s : SimpTheorem) : SimpM MessageData :=
+  s.toMessageDataM fun n => do
+    if (← getEnv).contains n then
+      return ← mkConstWithLevelParams n
+    if (← getLCtx).contains ⟨n⟩ then
```

Comment:
Good. I found the explanation for the change: free variable ids as theorem identifiers. We need to document this. Could you please document the field `name` and explain the new approach?